pkg/datapath/linux/probes/probes.go

```diff
@@ -217,3 +223,66 @@ func (p *ProbeManager) GetHelpers(prog string) map[string]struct{} {
 	}
 	return nil
 }
+
+// writeHeaders executes bpftool to generate BPF feature C macros and then
+// writes them to the given writer.
+func (p *ProbeManager) writeHeaders(featuresFile io.Writer) error {
+	cmd := exec.WithTimeout(
+		defaults.ExecTimeout, "bpftool", "feature", "probe", "macros")
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize stdout pipe for bpftool: %s", err)
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf(
+			"could not initialize stderr pipe for bpftool: %s", err)
+	}
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("could not start bpftool: %s", err)
+	}
+
+	writer := bufio.NewWriter(featuresFile)
+	defer writer.Flush()
+
+	io.WriteString(writer, "#ifndef BPF_FEATURES_H_\n")
+	io.WriteString(writer, "#define BPF_FEATURES_H_\n\n")
+
+	go io.Copy(writer, stdoutPipe)
+	if err := cmd.Wait(); err != nil {
+		stderr, err := ioutil.ReadAll(stderrPipe)
+		if err != nil {
+			return fmt.Errorf(
+				"bpftool failed: %s",
+				err)
+		}
+		return fmt.Errorf("bpftool did not run successfully: %s: %s",
+			err, stderr)
+	}
+
+	io.WriteString(writer, "#endif /* BPF_FEATURES_H_ */\n")
+
+	return nil
+}
+
+// CreateHeadersFile creates a C header file with macros indicating which BPF
+// features are available in the kernel.
+func (p *ProbeManager) CreateHeadersFile() error {
+	globalsDir := option.Config.GetGlobalsDir()
+	if err := os.MkdirAll(globalsDir, defaults.StateDirRights); err != nil && os.IsExist(err) {
```

Comment:
This error check doesn't seem right.

From the `os.MkdirAll()` documentation:

```
If path is already a directory, MkdirAll does nothing and returns nil.
```

Here we seem to be only checking for one error type and failing only on that error type.

I think we can just do this?
```suggestion
	if err := os.MkdirAll(globalsDir, defaults.StateDirRights); err != nil {
```